File: java-advent-of-code/src/test/java/advent/twenty_seventeen/Day04HighEntropyPassphrasesTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day04HighEntropyPassphrasesTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day04HighEntropyPassphrases("/twenty_seventeen/Day04-HighEntropyPassphrases-input.txt");

        assertEquals(386, dp.getPart1Answer().intValue());
        assertEquals(208, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day04HighEntropyPassphrases("/twenty_seventeen/Day04-HighEntropyPassphrases-example#1.txt");

        assertEquals(2, dp.getPart1Answer().intValue());
        assertEquals(2, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day04HighEntropyPassphrasesTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day04HighEntropyPassphrasesTest  extends FlatSpec with Matchers {
  "2017 Day 04 - Input" should "calculate answers" in {
    val dp = new Day04HighEntropyPassphrases("/twenty_seventeen/Day04-HighEntropyPassphrases-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day04HighEntropyPassphrases("/twenty_seventeen/Day04-HighEntropyPassphrases-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
